mlflow/transformers/__init__.py

```diff
@@ -1183,8 +1175,22 @@ def _load_model(path: str, flavor_config, return_type: str, device=None, **kwarg
         conf["device_map"] = device_map_strategy
         accelerate_model_conf["device_map"] = device_map_strategy
         # Cannot use device with device_map
+        if device is not None:
+            _logger.warning(
+                "MLFLOW_HUGGINGFACE_USE_DEVICE_MAP is set to True, device "
+                f"is incompatible with device_map, ignoring device param `{device}` passed."
```

Comment:
Shall we raise here? Now that L1170-1177 code block is moved to after this check, then `device` parameter is always opt-in.

Reply:
Why do we need to raise? Any concerns about warning and ignoring?

Reply:
Sure, here are the reasons I prefer exception:
1. It is incorrect usage of the API.
2. When users specify device, it means they want to load the model onto that specific devices. Ignoring it results in undesired outcome e.g. slow performance or OOM.
3. The priority between device map and device is not obvious to users. It is extra complexity we adds on top of transformer.

Reply:
+1 to raising. With a 1+ hour start-up time for many of these models in model serving, having a fast-fail for incompatible config is preferred. Can we make the exception message very clear about the incompatibility and how to fix it?